packages/protocol/contracts/team/TimelockTokenPool.sol

```diff
@@ -169,18 +174,25 @@ contract TimelockTokenPool is EssentialContract, EIP712Upgradeable {
 
     /// @notice Withdraws all withdrawable tokens to a designated address.
     /// @param _to The address where the granted and unlocked tokens shall be sent to.
+    /// @param _nonce The nonce to be used.
     /// @param _sig Signature provided by the grant recipient.
-    function withdraw(address _to, bytes memory _sig) external nonReentrant {
+    function withdraw(address _to, uint256 _nonce, bytes memory _sig) external nonReentrant {
         if (_to == address(0)) revert INVALID_PARAM();
-        address recipient = ECDSA.recover(getWithdrawalHash(_to), _sig);
-        _withdraw(recipient, _to);
+
+        address account = ECDSA.recover(getWithdrawalHash(_to, _nonce), _sig);
+        if (account == address(0)) revert INVALID_SIGNATURE();
+        if (withdrawalNonces[account] != _nonce) revert INVALID_NONCE();
+
+        withdrawalNonces[account] += 1;
+        _withdraw(account, _to);
     }
 
     /// @notice Gets the hash to be signed to authorize an withdrawal.
     /// @param _to The destination address.
+    /// @param _nonce The withdrawal nonce.
     /// @return The hash to be signed.
-    function getWithdrawalHash(address _to) public view returns (bytes32) {
-        return _hashTypedDataV4(keccak256(abi.encode(TYPED_HASH, _to)));
+    function getWithdrawalHash(address _to, uint256 _nonce) public view returns (bytes32) {
```

Comment:
to be cross-chain fixed, maybe block.chainId (?).

Reply:
its part of _hashTypedDataV4 I believe.

Reply:
> its part of _hashTypedDataV4 I believe.

Ah yeah, false alarm!